scripts/install/GenerisSearchWhitelist.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoPublishing\scripts\install;
+
+use oat\oatbox\extension\InstallAction;
+use oat\tao\model\search\SearchProxy;
+use oat\taoPublishing\model\PlatformService;
+
+class GenerisSearchWhitelist extends InstallAction
+{
+    public function __invoke($params)
+    {
+        /** @var SearchProxy $searchProxy */
+        $searchProxy = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
+
+        $generisSearchWhitelist = [
+            PlatformService::CLASS_URI
+        ];
+
+        if ($searchProxy->hasOption(SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST)) {
+            $options = $searchProxy->getOption(SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST);
+            $generisSearchWhitelist = array_merge($options, $generisSearchWhitelist);
+        }
+        $searchProxy->setOption(SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST, $generisSearchWhitelist);
+
+        $this->getServiceManager()->register(SearchProxy::SERVICE_ID, $searchProxy);
```

Comment:
https://github.com/oat-sa/extension-tao-publishing/pull/115/files#r695437048

Lets try to make code small instead of adding unnecessary ifs and temporary variables.

```suggestion
        /** @var SearchProxy $searchProxy */
        $searchProxy = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
        
        $searchProxy->setOption(
            SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST, 
            array_merge(
                $searchProxy->getOption(SearchProxy::OPTION_GENERIS_SEARCH_WHITELIST, []),
                [
                    PlatformService::CLASS_URI
                ]
            )
        );

        $this->getServiceManager()->register(SearchProxy::SERVICE_ID, $searchProxy);
```